<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>942</width>
    <height>550</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QGroupBox" name="keyBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>301</width>
      <height>451</height>
     </rect>
    </property>
    <property name="title">
     <string>Ключевая последовательность</string>
    </property>
    <widget class="QLabel" name="label_7">
     <property name="geometry">
      <rect>
       <x>13</x>
       <y>180</y>
       <width>211</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Длина последовательности:</string>
     </property>
    </widget>
    <widget class="QPlainTextEdit" name="showSeq">
     <property name="geometry">
      <rect>
       <x>13</x>
       <y>370</y>
       <width>276</width>
       <height>70</height>
      </rect>
     </property>
    </widget>
    <widget class="QSpinBox" name="regLength">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>133</x>
       <y>90</y>
       <width>71</width>
       <height>21</height>
      </rect>
     </property>
     <property name="maximum">
      <number>1000</number>
     </property>
     <property name="value">
      <number>100</number>
     </property>
    </widget>
    <widget class="QSpinBox" name="bRank">
     <property name="geometry">
      <rect>
       <x>153</x>
       <y>150</y>
       <width>71</width>
       <height>21</height>
      </rect>
     </property>
     <property name="maximum">
      <number>1000</number>
     </property>
     <property name="value">
      <number>1</number>
     </property>
    </widget>
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>13</x>
       <y>90</y>
       <width>121</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Длина регистра:</string>
     </property>
    </widget>
    <widget class="QPushButton" name="loadReg">
     <property name="geometry">
      <rect>
       <x>13</x>
       <y>50</y>
       <width>276</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>Загрузить регистр сдвига</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="aRank">
     <property name="geometry">
      <rect>
       <x>37</x>
       <y>150</y>
       <width>71</width>
       <height>21</height>
      </rect>
     </property>
     <property name="maximum">
      <number>1000</number>
     </property>
     <property name="value">
      <number>99</number>
     </property>
    </widget>
    <widget class="QLabel" name="label_5">
     <property name="geometry">
      <rect>
       <x>13</x>
       <y>340</y>
       <width>121</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Вывести первые </string>
     </property>
    </widget>
    <widget class="QLabel" name="label_6">
     <property name="geometry">
      <rect>
       <x>213</x>
       <y>340</y>
       <width>61</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>членов:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_3">
     <property name="geometry">
      <rect>
       <x>13</x>
       <y>150</y>
       <width>21</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>a =</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_4">
     <property name="geometry">
      <rect>
       <x>128</x>
       <y>150</y>
       <width>21</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>b =</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="lengthOfShow">
     <property name="geometry">
      <rect>
       <x>135</x>
       <y>340</y>
       <width>71</width>
       <height>21</height>
      </rect>
     </property>
     <property name="maximum">
      <number>100000</number>
     </property>
    </widget>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>13</x>
       <y>120</y>
       <width>271</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Разряды для суммирования:</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="seqLength">
     <property name="geometry">
      <rect>
       <x>218</x>
       <y>185</y>
       <width>71</width>
       <height>21</height>
      </rect>
     </property>
     <property name="maximum">
      <number>1000000</number>
     </property>
     <property name="value">
      <number>10000</number>
     </property>
    </widget>
    <widget class="QPushButton" name="genSeq">
     <property name="geometry">
      <rect>
       <x>13</x>
       <y>300</y>
       <width>276</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>Сгенерировать последовательность</string>
     </property>
    </widget>
    <widget class="QPushButton" name="savReg">
     <property name="geometry">
      <rect>
       <x>13</x>
       <y>260</y>
       <width>276</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>Сохранить регистр сдвига</string>
     </property>
    </widget>
    <widget class="QPushButton" name="genReg">
     <property name="geometry">
      <rect>
       <x>13</x>
       <y>220</y>
       <width>276</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string> Сгенерировать регистр сдвига</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="serialBox">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>10</y>
      <width>301</width>
      <height>221</height>
     </rect>
    </property>
    <property name="title">
     <string>Сериальный тест</string>
    </property>
    <widget class="QLabel" name="label_8">
     <property name="geometry">
      <rect>
       <x>13</x>
       <y>80</y>
       <width>101</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Длина серии:</string>
     </property>
    </widget>
    <widget class="QComboBox" name="serialComboBox">
     <property name="geometry">
      <rect>
       <x>115</x>
       <y>80</y>
       <width>51</width>
       <height>21</height>
      </rect>
     </property>
     <property name="currentText">
      <string/>
     </property>
     <property name="currentIndex">
      <number>-1</number>
     </property>
     <property name="maxVisibleItems">
      <number>3</number>
     </property>
     <property name="maxCount">
      <number>3</number>
     </property>
    </widget>
    <widget class="QPushButton" name="serialTest">
     <property name="geometry">
      <rect>
       <x>13</x>
       <y>140</y>
       <width>276</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>Тестировать</string>
     </property>
    </widget>
    <widget class="QLabel" name="kritPyrson">
     <property name="geometry">
      <rect>
       <x>13</x>
       <y>180</y>
       <width>221</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Критерий Пирсона = 0</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_10">
     <property name="geometry">
      <rect>
       <x>230</x>
       <y>130</y>
       <width>51</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLabel" name="pyrsonVal">
     <property name="geometry">
      <rect>
       <x>13</x>
       <y>110</y>
       <width>271</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Допустимые значения: 0.352÷7.815</string>
     </property>
    </widget>
    <widget class="QRadioButton" name="textRadioButtonForSerial">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>50</y>
       <width>112</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Текст</string>
     </property>
    </widget>
    <widget class="QRadioButton" name="keyRadioButtonForSerial">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>13</x>
       <y>50</y>
       <width>112</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Ключ</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
    <zorder>serialComboBox</zorder>
    <zorder>label_8</zorder>
    <zorder>serialTest</zorder>
    <zorder>kritPyrson</zorder>
    <zorder>label_10</zorder>
    <zorder>pyrsonVal</zorder>
    <zorder>textRadioButtonForSerial</zorder>
    <zorder>keyRadioButtonForSerial</zorder>
   </widget>
   <widget class="QGroupBox" name="correlBox">
    <property name="geometry">
     <rect>
      <x>630</x>
      <y>10</y>
      <width>301</width>
      <height>221</height>
     </rect>
    </property>
    <property name="title">
     <string>Корреляционный тест</string>
    </property>
    <widget class="QLabel" name="label_9">
     <property name="geometry">
      <rect>
       <x>13</x>
       <y>80</y>
       <width>81</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Смещение:</string>
     </property>
    </widget>
    <widget class="QComboBox" name="correlComboBox">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>80</y>
       <width>51</width>
       <height>21</height>
      </rect>
     </property>
     <property name="currentText">
      <string/>
     </property>
     <property name="currentIndex">
      <number>-1</number>
     </property>
     <property name="maxVisibleItems">
      <number>4</number>
     </property>
     <property name="maxCount">
      <number>4</number>
     </property>
    </widget>
    <widget class="QPushButton" name="correlTest">
     <property name="geometry">
      <rect>
       <x>13</x>
       <y>140</y>
       <width>271</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>Тестировать</string>
     </property>
    </widget>
    <widget class="QLabel" name="correlLabel">
     <property name="geometry">
      <rect>
       <x>13</x>
       <y>180</y>
       <width>271</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string> Коэф. автокорреляции = 0</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_11">
     <property name="geometry">
      <rect>
       <x>230</x>
       <y>130</y>
       <width>51</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLabel" name="correlVal">
     <property name="geometry">
      <rect>
       <x>13</x>
       <y>110</y>
       <width>281</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Допустимые значения: 0</string>
     </property>
    </widget>
    <widget class="QRadioButton" name="textRadioButtonForCorrel">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>50</y>
       <width>112</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Текст</string>
     </property>
    </widget>
    <widget class="QRadioButton" name="keyRadioButtonForCorrel">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>13</x>
       <y>50</y>
       <width>112</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Ключ</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="codeBox">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>240</y>
      <width>611</width>
      <height>221</height>
     </rect>
    </property>
    <property name="title">
     <string>Шифрование/дешифрование</string>
    </property>
    <widget class="QPushButton" name="loadText">
     <property name="geometry">
      <rect>
       <x>13</x>
       <y>50</y>
       <width>141</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>Выбрать файл</string>
     </property>
    </widget>
    <widget class="QPushButton" name="codeText">
     <property name="geometry">
      <rect>
       <x>13</x>
       <y>90</y>
       <width>141</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>Шифровать</string>
     </property>
    </widget>
    <widget class="QPushButton" name="decodeText">
     <property name="geometry">
      <rect>
       <x>13</x>
       <y>130</y>
       <width>141</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>Дешифровать</string>
     </property>
    </widget>
    <widget class="QPlainTextEdit" name="text">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>50</y>
       <width>441</width>
       <height>141</height>
      </rect>
     </property>
    </widget>
    <widget class="QRadioButton" name="keyButton">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>13</x>
       <y>-160</y>
       <width>112</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Ключ</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>942</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
